src/NewsFeed/NewsFeed.tsx

```diff
@@ -72,31 +74,29 @@ function NewsFeed() {
       })
       .then(
         () => {
-          // supposed to sort stories by highest score
+          // sort stories by highest score
           randomStoriesContent.sort((a: NewsFeedStory, b: NewsFeedStory) =>
             a.score < b.score ? 1 : b.score < a.score ? -1 : 0
           );
 
-          console.log("After sort: ", randomStoriesContent);
-          setLoading(2);
+          setLoading(0);
         },
         (error) => {
           console.log("Could not sort stories. ", error);
         }
       );
   }, []); // Empty dependency array to run effect only once on component mount
 
-  if (!hasLoaded) {
+  if (isLoading) {
```

Comment:
If `isLoading` is 1 or 2 then this statement will be true and will execute its body, the statement will never reach to `isLoading` === 1.